sympy/physics/control/lti.py

```diff
@@ -3838,6 +3838,107 @@ def num_outputs(self):
         """
         return self._D.rows
 
+    def state_vector(self, initial_conditions=None, input_vector=None, var=None):
+        r"""
+        Returns the State vector `x` given by the solution of the state equation
+            x'(t) = A * x(t) + B * u(t)
+
+        Parameters
+        ============
+
+        initial_conditions : Matrix
+            The initial conditions of `x` state vector.
+        input_vector : Matrix
+            The input vector for state space.
+        var : Symbol
+            The symbol representing time.
+
+        Examples
+        ==========
+
+        >>> from sympy import Matrix, Symbol
+        >>> from sympy.physics.control import StateSpace
+        >>> A = Matrix([[-2, 0], [1, -1]])
+        >>> i = Matrix([2, 3])
+        >>> t = Symbol('t')
+        >>> ss = StateSpace(A)
+        >>> ss.state_vector(initial_conditions=i, var=t)
+        Matrix([
+        [            2*exp(-2*t)],
+        [5*exp(-t) - 2*exp(-2*t)]])
+
+        References
+        ==========
+        .. [1] https://web.mit.edu/2.14/www/Handouts/StateSpaceResponse.pdf
+        .. [2] https://docs.sympy.org/latest/modules/solvers/ode.html#sympy.solvers.ode.systems.linodesolve
+        """
+
+        from sympy.solvers.ode.systems import linodesolve
+        from sympy import Eq, solve
+
+        if not var:
+            var = Symbol('t')
+        elif not isinstance(var, Symbol):
+            raise ValueError("Variable for representing time must be a Symbol.")
+        if not initial_conditions:
+            initial_conditions = zeros(self._A.shape[0], 1)
+        elif initial_conditions.shape != (self._A.shape[0], 1):
+            raise ShapeError("Initial condition vector should have the same number of "
+                             "rows as the state matrix.")
+        if not input_vector:
+            input_vector = zeros(self._B.shape[1], 1)
+        elif input_vector.shape != (self._B.shape[1], 1):
+            raise ShapeError("Input vector should have the same number of "
+                             "columns as the input matrix.")
+        sol = linodesolve(self._A, var, self._B*input_vector, type='type2', doit=True)
+        res = Matrix(sol)
+        mat2 = res.replace(var, 0)
+        eqn = Eq(mat2, initial_conditions)
+        sol_eqn = solve(eqn)
```

Comment:
Having not followed the recent development in sympy solvers I feel a bit difficult to comment if this is optimal. Why is `solveset` not used here as well ? -https://docs.sympy.org/latest/modules/solvers/solveset.html#what-s-wrong-with-solve

Reply:
I don't think that it is possible to use solveset, because the equations contain matrix, however, it may be better to use linsolve

Reply:
Thank you for letting me know about this. I've used `linsolve` in the latest commit.